{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c77f8da1-2837-4bf7-b3f4-dfcaa7ab42f6",
   "metadata": {
    "execution": {
     "iopub.execute_input": "2023-11-08T19:47:20.262444Z",
     "iopub.status.busy": "2023-11-08T19:47:20.262325Z",
     "iopub.status.idle": "2023-11-08T19:47:20.344406Z",
     "shell.execute_reply": "2023-11-08T19:47:20.343863Z"
    }
   },
   "outputs": [],
   "source": [
    "import scipy.io\n",
    "import numpy as np\n",
    "import random\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3fe25376-b856-4296-ba02-5e3771ff9451",
   "metadata": {
    "collapsed": true,
    "execution": {
     "iopub.execute_input": "2023-11-08T19:47:20.347040Z",
     "iopub.status.busy": "2023-11-08T19:47:20.346874Z",
     "iopub.status.idle": "2023-11-08T21:37:38.706785Z",
     "shell.execute_reply": "2023-11-08T21:37:38.706231Z"
    },
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:47:20.527650: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
      "To enable the following instructions: FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
      "/usr/lib/python3.11/site-packages/h5py/__init__.py:36: UserWarning: h5py is running against HDF5 1.14.2 when it was built against 1.14.1, this may cause problems\n",
      "  _warn((\"h5py is running against HDF5 {0} when it was built against {1}, \"\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "35227\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/compat/v2_compat.py:107: disable_resource_variables (from tensorflow.python.ops.variable_scope) is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "non-resource variables are not supported in the long term\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:46.113429: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.125301: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.125957: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.129552: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.133214: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.133386: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.213321: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.213525: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.213691: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:46.213834: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 14:48:46.214256: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:47.151783: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.152079: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.152283: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.152518: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.152712: I tensorflow/compiler/xla/stream_executor/cuda/cuda_gpu_executor.cc:995] successful NUMA node read from SysFS had negative value (-1), but there must be at least one NUMA node, so returning NUMA node zero. See more at https://github.com/torvalds/linux/blob/v6.0/Documentation/ABI/testing/sysfs-bus-pci#L344-L355\n",
      "2023-11-08 14:48:47.152911: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1639] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 46608 MB memory:  -> device: 0, name: NVIDIA RTX A6000, pci bus id: 0000:2d:00.0, compute capability: 8.6\n",
      "2023-11-08 14:48:47.152948: I tensorflow/core/common_runtime/process_util.cc:146] Creating new thread pool with default inter op setting: 2. Tune using inter_op_parallelism_threads for best performance.\n",
      "2023-11-08 14:48:47.178792: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:375] MLIR V1 optimization pass is not enabled\n",
      "2023-11-08 14:48:47.403167: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_21/lstm_cell_21/bias/Assign' id:3510 op device:{requested: '', assigned: ''} def:{{{node lstm_21/lstm_cell_21/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_21/lstm_cell_21/bias, lstm_21/lstm_cell_21/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:48:47.538770: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_1' id:3634 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_1}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 0 0 0...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 14:48:47.572343: W tensorflow/c/c_api.cc:304] Operation '{name:'strided_slice/stack_2' id:3635 op device:{requested: '', assigned: ''} def:{{{node strided_slice/stack_2}} = Const[_has_manual_control_dependencies=true, dtype=DT_INT32, value=Tensor<type: int32 shape: [4] values: 1 1 1...>]()}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "(31688, 95)\n",
      "Train on 31688 samples, validate on 3539 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:51.987221: W tensorflow/c/c_api.cc:304] Operation '{name:'training/Adam/conv2d_1/bias/v/Assign' id:17012 op device:{requested: '', assigned: ''} def:{{{node training/Adam/conv2d_1/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](training/Adam/conv2d_1/bias/v, training/Adam/conv2d_1/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/50\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 14:48:55.763304: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n",
      "2023-11-08 14:48:57.874922: I tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:606] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.\n",
      "2023-11-08 14:48:57.884646: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:432] Loaded cuDNN version 8902\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31688/31688 [==============================] - ETA: 0s - loss: 3.0524"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2335: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates = self.state_updates\n",
      "2023-11-08 14:49:19.035459: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_1/mul' id:6477 op device:{requested: '', assigned: ''} def:{{{node loss_1/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_1/mul/x, loss_1/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.84284, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 28s 895us/sample - loss: 3.0524 - val_loss: 1.8428\n",
      "Epoch 2/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.7579\n",
      "Epoch 2: val_loss improved from 1.84284 to 1.58444, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.7579 - val_loss: 1.5844\n",
      "Epoch 3/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.5655\n",
      "Epoch 3: val_loss improved from 1.58444 to 1.52718, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 686us/sample - loss: 1.5655 - val_loss: 1.5272\n",
      "Epoch 4/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.5137\n",
      "Epoch 4: val_loss improved from 1.52718 to 1.50536, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 609us/sample - loss: 1.5137 - val_loss: 1.5054\n",
      "Epoch 5/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4895\n",
      "Epoch 5: val_loss improved from 1.50536 to 1.49726, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 600us/sample - loss: 1.4895 - val_loss: 1.4973\n",
      "Epoch 6/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4789\n",
      "Epoch 6: val_loss improved from 1.49726 to 1.48195, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 602us/sample - loss: 1.4789 - val_loss: 1.4820\n",
      "Epoch 7/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4619\n",
      "Epoch 7: val_loss improved from 1.48195 to 1.48127, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 598us/sample - loss: 1.4619 - val_loss: 1.4813\n",
      "Epoch 8/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4603\n",
      "Epoch 8: val_loss improved from 1.48127 to 1.47299, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 677us/sample - loss: 1.4603 - val_loss: 1.4730\n",
      "Epoch 9/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4534\n",
      "Epoch 9: val_loss improved from 1.47299 to 1.47010, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 677us/sample - loss: 1.4534 - val_loss: 1.4701\n",
      "Epoch 10/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4478\n",
      "Epoch 10: val_loss improved from 1.47010 to 1.46209, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 606us/sample - loss: 1.4478 - val_loss: 1.4621\n",
      "Epoch 11/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4438\n",
      "Epoch 11: val_loss improved from 1.46209 to 1.45830, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.4438 - val_loss: 1.4583\n",
      "Epoch 12/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4399\n",
      "Epoch 12: val_loss improved from 1.45830 to 1.45077, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 601us/sample - loss: 1.4399 - val_loss: 1.4508\n",
      "Epoch 13/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4392\n",
      "Epoch 13: val_loss improved from 1.45077 to 1.44914, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 599us/sample - loss: 1.4392 - val_loss: 1.4491\n",
      "Epoch 14/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4399\n",
      "Epoch 14: val_loss did not improve from 1.44914\n",
      "31688/31688 [==============================] - 21s 650us/sample - loss: 1.4399 - val_loss: 1.4575\n",
      "Epoch 15/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4322\n",
      "Epoch 15: val_loss did not improve from 1.44914\n",
      "31688/31688 [==============================] - 20s 647us/sample - loss: 1.4322 - val_loss: 1.4538\n",
      "Epoch 16/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4353\n",
      "Epoch 16: val_loss improved from 1.44914 to 1.44692, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.4353 - val_loss: 1.4469\n",
      "Epoch 17/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4427\n",
      "Epoch 17: val_loss did not improve from 1.44692\n",
      "31688/31688 [==============================] - 22s 689us/sample - loss: 1.4427 - val_loss: 1.4544\n",
      "Epoch 18/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4298\n",
      "Epoch 18: val_loss improved from 1.44692 to 1.44213, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.4298 - val_loss: 1.4421\n",
      "Epoch 19/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4445\n",
      "Epoch 19: val_loss did not improve from 1.44213\n",
      "31688/31688 [==============================] - 20s 635us/sample - loss: 1.4445 - val_loss: 1.4454\n",
      "Epoch 20/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4285\n",
      "Epoch 20: val_loss improved from 1.44213 to 1.43931, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 605us/sample - loss: 1.4285 - val_loss: 1.4393\n",
      "Epoch 21/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4292\n",
      "Epoch 21: val_loss improved from 1.43931 to 1.43569, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 604us/sample - loss: 1.4292 - val_loss: 1.4357\n",
      "Epoch 22/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4356\n",
      "Epoch 22: val_loss did not improve from 1.43569\n",
      "31688/31688 [==============================] - 21s 665us/sample - loss: 1.4356 - val_loss: 1.4395\n",
      "Epoch 23/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4202\n",
      "Epoch 23: val_loss did not improve from 1.43569\n",
      "31688/31688 [==============================] - 21s 667us/sample - loss: 1.4202 - val_loss: 1.4399\n",
      "Epoch 24/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4139\n",
      "Epoch 24: val_loss improved from 1.43569 to 1.42939, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 604us/sample - loss: 1.4139 - val_loss: 1.4294\n",
      "Epoch 25/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4097\n",
      "Epoch 25: val_loss did not improve from 1.42939\n",
      "31688/31688 [==============================] - 19s 597us/sample - loss: 1.4097 - val_loss: 1.4315\n",
      "Epoch 26/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4092\n",
      "Epoch 26: val_loss did not improve from 1.42939\n",
      "31688/31688 [==============================] - 20s 625us/sample - loss: 1.4092 - val_loss: 1.4307\n",
      "Epoch 27/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4044\n",
      "Epoch 27: val_loss improved from 1.42939 to 1.42017, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 665us/sample - loss: 1.4044 - val_loss: 1.4202\n",
      "Epoch 28/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4014\n",
      "Epoch 28: val_loss did not improve from 1.42017\n",
      "31688/31688 [==============================] - 22s 682us/sample - loss: 1.4014 - val_loss: 1.4221\n",
      "Epoch 29/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3979\n",
      "Epoch 29: val_loss improved from 1.42017 to 1.41956, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.3979 - val_loss: 1.4196\n",
      "Epoch 30/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3962\n",
      "Epoch 30: val_loss improved from 1.41956 to 1.41716, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 660us/sample - loss: 1.3962 - val_loss: 1.4172\n",
      "Epoch 31/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3929\n",
      "Epoch 31: val_loss improved from 1.41716 to 1.41440, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 611us/sample - loss: 1.3929 - val_loss: 1.4144\n",
      "Epoch 32/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3903\n",
      "Epoch 32: val_loss did not improve from 1.41440\n",
      "31688/31688 [==============================] - 19s 610us/sample - loss: 1.3903 - val_loss: 1.4230\n",
      "Epoch 33/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3879\n",
      "Epoch 33: val_loss improved from 1.41440 to 1.41412, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 695us/sample - loss: 1.3879 - val_loss: 1.4141\n",
      "Epoch 34/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3899\n",
      "Epoch 34: val_loss improved from 1.41412 to 1.40769, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 689us/sample - loss: 1.3899 - val_loss: 1.4077\n",
      "Epoch 35/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3867\n",
      "Epoch 35: val_loss did not improve from 1.40769\n",
      "31688/31688 [==============================] - 21s 669us/sample - loss: 1.3867 - val_loss: 1.4097\n",
      "Epoch 36/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3840\n",
      "Epoch 36: val_loss did not improve from 1.40769\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 1.3840 - val_loss: 1.4162\n",
      "Epoch 37/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3825\n",
      "Epoch 37: val_loss improved from 1.40769 to 1.40537, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 1.3825 - val_loss: 1.4054\n",
      "Epoch 38/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3817\n",
      "Epoch 38: val_loss improved from 1.40537 to 1.40337, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 20s 622us/sample - loss: 1.3817 - val_loss: 1.4034\n",
      "Epoch 39/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3784\n",
      "Epoch 39: val_loss did not improve from 1.40337\n",
      "31688/31688 [==============================] - 19s 601us/sample - loss: 1.3784 - val_loss: 1.4061\n",
      "Epoch 40/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3748\n",
      "Epoch 40: val_loss improved from 1.40337 to 1.40247, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 667us/sample - loss: 1.3748 - val_loss: 1.4025\n",
      "Epoch 41/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3729\n",
      "Epoch 41: val_loss did not improve from 1.40247\n",
      "31688/31688 [==============================] - 22s 687us/sample - loss: 1.3729 - val_loss: 1.4025\n",
      "Epoch 42/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3719\n",
      "Epoch 42: val_loss improved from 1.40247 to 1.40140, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 1.3719 - val_loss: 1.4014\n",
      "Epoch 43/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3708\n",
      "Epoch 43: val_loss improved from 1.40140 to 1.39734, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3708 - val_loss: 1.3973\n",
      "Epoch 44/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3714\n",
      "Epoch 44: val_loss improved from 1.39734 to 1.39441, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 661us/sample - loss: 1.3714 - val_loss: 1.3944\n",
      "Epoch 45/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3680\n",
      "Epoch 45: val_loss did not improve from 1.39441\n",
      "31688/31688 [==============================] - 22s 684us/sample - loss: 1.3680 - val_loss: 1.3968\n",
      "Epoch 46/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3669\n",
      "Epoch 46: val_loss did not improve from 1.39441\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3669 - val_loss: 1.3950\n",
      "Epoch 47/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3628\n",
      "Epoch 47: val_loss did not improve from 1.39441\n",
      "31688/31688 [==============================] - 22s 680us/sample - loss: 1.3628 - val_loss: 1.3962\n",
      "Epoch 48/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3657\n",
      "Epoch 48: val_loss did not improve from 1.39441\n",
      "31688/31688 [==============================] - 21s 669us/sample - loss: 1.3657 - val_loss: 1.3959\n",
      "Epoch 49/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3617\n",
      "Epoch 49: val_loss improved from 1.39441 to 1.39235, saving model to ./checkpoints/unknown_person_baseline_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 683us/sample - loss: 1.3617 - val_loss: 1.3923\n",
      "Epoch 50/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3618\n",
      "Epoch 50: val_loss did not improve from 1.39235\n",
      "31688/31688 [==============================] - 19s 605us/sample - loss: 1.3618 - val_loss: 1.4002\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling GlorotUniform.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Zeros.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:From /usr/lib/python3.11/site-packages/tensorflow/python/ops/init_ops.py:94: calling Orthogonal.__init__ (from tensorflow.python.ops.init_ops) with dtype is deprecated and will be removed in a future version.\n",
      "Instructions for updating:\n",
      "Call initializer instance with the dtype argument instead of passing it to the constructor\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:06:12.951921: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_24_1/lstm_cell_61/bias/Assign' id:22511 op device:{requested: '', assigned: ''} def:{{{node lstm_24_1/lstm_cell_61/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_24_1/lstm_cell_61/bias, lstm_24_1/lstm_cell_61/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:06:14.683896: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_35_1/lstm_cell_72/kernel/v/Assign' id:26420 op device:{requested: '', assigned: ''} def:{{{node lstm_35_1/lstm_cell_72/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_35_1/lstm_cell_72/kernel/v, lstm_35_1/lstm_cell_72/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "/usr/lib/python3.11/site-packages/keras/engine/training_v1.py:2359: UserWarning: `Model.state_updates` will be removed in a future version. This property should not be used in TensorFlow 2.0, as `updates` are applied automatically.\n",
      "  updates=self.state_updates,\n",
      "2023-11-08 15:06:16.021693: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_1/cond/Merge' id:24591 op device:{requested: '', assigned: ''} def:{{{node dropout_4_1/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_1/cond/Identity, dropout_4_1/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1774)\n",
      "(1514, 1774)\n",
      "(1644, 1774)\n",
      "(1764, 1774)\n",
      "(1836, 1774)\n",
      "(1699, 1774)\n",
      "(1369, 1774)\n",
      "(1778, 1774)\n",
      "(1788, 1774)\n",
      "(1538, 1774)\n",
      "(1908, 1774)\n",
      "(1691, 1774)\n",
      "(1800, 1774)\n",
      "(1848, 1774)\n",
      "(1704, 1774)\n",
      "(1824, 1774)\n",
      "(958, 1774)\n",
      "(1680, 1774)\n",
      "(1860, 1774)\n",
      "{1: 5.90766581284347, 2: 5.8821380418739775, 4: 10.0, 5: 5.277217057677138, 6: 5.390831521536593, 8: 8.86656223783471, 9: 7.057637149203656, 10: 7.347788281260194, 12: 8.72292881851828, 13: 6.116892829527895, 17: 9.357382128701667, 19: 8.439122456119566, 21: 8.591293857082068, 22: 1.0, 25: 7.605661348130724, 26: 6.029527632415268, 27: 5.661130536759628, 28: 5.706057588805419, 29: 2.82438148821693}\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/ipykernel_2792526/2970945921.py:300: DeprecationWarning: Conversion of an array with ndim > 0 to a scalar is deprecated, and will error in future. Ensure you extract a single element from your array before performing this operation. (Deprecated NumPy 1.25.)\n",
      "  w_train[i] = weights_dict[int(p_train[i])]\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31688 samples, validate on 3539 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:09:53.758160: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_5/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31688/31688 [==============================] - ETA: 0s - loss: 9.4291\n",
      "Epoch 1: val_loss improved from inf to 1.41627, saving model to ./checkpoints/unknown_person_p11_20.h5\n",
      "31688/31688 [==============================] - 25s 800us/sample - loss: 9.4291 - val_loss: 1.4163\n",
      "Epoch 2/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.3577\n",
      "Epoch 2: val_loss improved from 1.41627 to 1.39545, saving model to ./checkpoints/unknown_person_p11_20.h5\n",
      "31688/31688 [==============================] - 20s 636us/sample - loss: 9.3577 - val_loss: 1.3954\n",
      "Epoch 3/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.3110\n",
      "Epoch 3: val_loss did not improve from 1.39545\n",
      "31688/31688 [==============================] - 20s 625us/sample - loss: 9.3110 - val_loss: 1.4110\n",
      "Epoch 4/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.2648\n",
      "Epoch 4: val_loss improved from 1.39545 to 1.38908, saving model to ./checkpoints/unknown_person_p11_20.h5\n",
      "31688/31688 [==============================] - 19s 606us/sample - loss: 9.2648 - val_loss: 1.3891\n",
      "Epoch 5/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.2583\n",
      "Epoch 5: val_loss did not improve from 1.38908\n",
      "31688/31688 [==============================] - 19s 609us/sample - loss: 9.2583 - val_loss: 1.3918\n",
      "Epoch 6/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.2468\n",
      "Epoch 6: val_loss did not improve from 1.38908\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 9.2468 - val_loss: 1.4003\n",
      "Epoch 7/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.2300\n",
      "Epoch 7: val_loss did not improve from 1.38908\n",
      "31688/31688 [==============================] - 19s 606us/sample - loss: 9.2300 - val_loss: 1.3918\n",
      "Epoch 8/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1938\n",
      "Epoch 8: val_loss did not improve from 1.38908\n",
      "31688/31688 [==============================] - 19s 612us/sample - loss: 9.1938 - val_loss: 1.4022\n",
      "Epoch 9/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1822\n",
      "Epoch 9: val_loss did not improve from 1.38908\n",
      "31688/31688 [==============================] - 19s 608us/sample - loss: 9.1822 - val_loss: 1.3941\n",
      "Epoch 10/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1525\n",
      "Epoch 10: val_loss did not improve from 1.38908\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 9.1525 - val_loss: 1.3893\n",
      "Epoch 11/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1466\n",
      "Epoch 11: val_loss improved from 1.38908 to 1.38842, saving model to ./checkpoints/unknown_person_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 677us/sample - loss: 9.1466 - val_loss: 1.3884\n",
      "Epoch 12/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1299\n",
      "Epoch 12: val_loss improved from 1.38842 to 1.38405, saving model to ./checkpoints/unknown_person_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 691us/sample - loss: 9.1299 - val_loss: 1.3840\n",
      "Epoch 13/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1089\n",
      "Epoch 13: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 22s 684us/sample - loss: 9.1089 - val_loss: 1.3850\n",
      "Epoch 14/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1255\n",
      "Epoch 14: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 21s 657us/sample - loss: 9.1255 - val_loss: 1.3885\n",
      "Epoch 15/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1033\n",
      "Epoch 15: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 19s 604us/sample - loss: 9.1033 - val_loss: 1.3890\n",
      "Epoch 16/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0946\n",
      "Epoch 16: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 20s 632us/sample - loss: 9.0946 - val_loss: 1.3891\n",
      "Epoch 17/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0993\n",
      "Epoch 17: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 9.0993 - val_loss: 1.3862\n",
      "Epoch 18/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0646\n",
      "Epoch 18: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 9.0646 - val_loss: 1.3841\n",
      "Epoch 19/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0593\n",
      "Epoch 19: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 22s 684us/sample - loss: 9.0593 - val_loss: 1.3890\n",
      "Epoch 20/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0504\n",
      "Epoch 20: val_loss did not improve from 1.38405\n",
      "31688/31688 [==============================] - 22s 686us/sample - loss: 9.0504 - val_loss: 1.3847\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:16:47.683279: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_28_2/lstm_cell_102/kernel/Assign' id:42520 op device:{requested: '', assigned: ''} def:{{{node lstm_28_2/lstm_cell_102/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_28_2/lstm_cell_102/kernel, lstm_28_2/lstm_cell_102/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:16:50.113134: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_31_2/lstm_cell_105/bias/v/Assign' id:45768 op device:{requested: '', assigned: ''} def:{{{node lstm_31_2/lstm_cell_105/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_31_2/lstm_cell_105/bias/v, lstm_31_2/lstm_cell_105/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31688 samples, validate on 3539 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:16:55.252020: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:17:07.902040: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_6/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3596"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:17:28.822151: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_5/mul' id:44579 op device:{requested: '', assigned: ''} def:{{{node loss_5/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_5/mul/x, loss_5/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.39310, saving model to ./checkpoints/unknown_person_baseline_2_p11_20.h5\n",
      "31688/31688 [==============================] - 27s 855us/sample - loss: 1.3596 - val_loss: 1.3931\n",
      "Epoch 2/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3591\n",
      "Epoch 2: val_loss improved from 1.39310 to 1.39099, saving model to ./checkpoints/unknown_person_baseline_2_p11_20.h5\n",
      "31688/31688 [==============================] - 20s 631us/sample - loss: 1.3591 - val_loss: 1.3910\n",
      "Epoch 3/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3590\n",
      "Epoch 3: val_loss did not improve from 1.39099\n",
      "31688/31688 [==============================] - 20s 622us/sample - loss: 1.3590 - val_loss: 1.3935\n",
      "Epoch 4/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3554\n",
      "Epoch 4: val_loss did not improve from 1.39099\n",
      "31688/31688 [==============================] - 20s 619us/sample - loss: 1.3554 - val_loss: 1.3966\n",
      "Epoch 5/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3557\n",
      "Epoch 5: val_loss improved from 1.39099 to 1.38533, saving model to ./checkpoints/unknown_person_baseline_2_p11_20.h5\n",
      "31688/31688 [==============================] - 21s 675us/sample - loss: 1.3557 - val_loss: 1.3853\n",
      "Epoch 6/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3559\n",
      "Epoch 6: val_loss did not improve from 1.38533\n",
      "31688/31688 [==============================] - 19s 610us/sample - loss: 1.3559 - val_loss: 1.3862\n",
      "Epoch 7/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3483\n",
      "Epoch 7: val_loss improved from 1.38533 to 1.38017, saving model to ./checkpoints/unknown_person_baseline_2_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3483 - val_loss: 1.3802\n",
      "Epoch 8/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3503\n",
      "Epoch 8: val_loss did not improve from 1.38017\n",
      "31688/31688 [==============================] - 22s 695us/sample - loss: 1.3503 - val_loss: 1.3883\n",
      "Epoch 9/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3491\n",
      "Epoch 9: val_loss did not improve from 1.38017\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3491 - val_loss: 1.3838\n",
      "Epoch 10/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3465\n",
      "Epoch 10: val_loss improved from 1.38017 to 1.37950, saving model to ./checkpoints/unknown_person_baseline_2_p11_20.h5\n",
      "31688/31688 [==============================] - 20s 628us/sample - loss: 1.3465 - val_loss: 1.3795\n",
      "Epoch 11/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3466\n",
      "Epoch 11: val_loss did not improve from 1.37950\n",
      "31688/31688 [==============================] - 22s 687us/sample - loss: 1.3466 - val_loss: 1.3895\n",
      "Epoch 12/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3444\n",
      "Epoch 12: val_loss did not improve from 1.37950\n",
      "31688/31688 [==============================] - 22s 692us/sample - loss: 1.3444 - val_loss: 1.3861\n",
      "Epoch 13/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3444\n",
      "Epoch 13: val_loss did not improve from 1.37950\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.3444 - val_loss: 1.3832\n",
      "Epoch 14/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3409\n",
      "Epoch 14: val_loss improved from 1.37950 to 1.37421, saving model to ./checkpoints/unknown_person_baseline_2_p11_20.h5\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 1.3409 - val_loss: 1.3742\n",
      "Epoch 15/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3422\n",
      "Epoch 15: val_loss did not improve from 1.37421\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.3422 - val_loss: 1.3815\n",
      "Epoch 16/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3428\n",
      "Epoch 16: val_loss did not improve from 1.37421\n",
      "31688/31688 [==============================] - 22s 687us/sample - loss: 1.3428 - val_loss: 1.3797\n",
      "Epoch 17/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3368\n",
      "Epoch 17: val_loss did not improve from 1.37421\n",
      "31688/31688 [==============================] - 21s 670us/sample - loss: 1.3368 - val_loss: 1.3769\n",
      "Epoch 18/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3393\n",
      "Epoch 18: val_loss did not improve from 1.37421\n",
      "31688/31688 [==============================] - 22s 687us/sample - loss: 1.3393 - val_loss: 1.3774\n",
      "Epoch 19/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3366\n",
      "Epoch 19: val_loss did not improve from 1.37421\n",
      "31688/31688 [==============================] - 22s 686us/sample - loss: 1.3366 - val_loss: 1.3762\n",
      "Epoch 20/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3363\n",
      "Epoch 20: val_loss did not improve from 1.37421\n",
      "31688/31688 [==============================] - 21s 675us/sample - loss: 1.3363 - val_loss: 1.3820\n",
      "Train on 31688 samples, validate on 3539 samples\n",
      "Epoch 1/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3363\n",
      "Epoch 1: val_loss improved from inf to 1.37849, saving model to ./checkpoints/unknown_person_baseline_p11_21.h5\n",
      "31688/31688 [==============================] - 21s 655us/sample - loss: 1.3363 - val_loss: 1.3785\n",
      "Epoch 2/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3357\n",
      "Epoch 2: val_loss improved from 1.37849 to 1.37462, saving model to ./checkpoints/unknown_person_baseline_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 697us/sample - loss: 1.3357 - val_loss: 1.3746\n",
      "Epoch 3/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3306\n",
      "Epoch 3: val_loss improved from 1.37462 to 1.37211, saving model to ./checkpoints/unknown_person_baseline_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 698us/sample - loss: 1.3306 - val_loss: 1.3721\n",
      "Epoch 4/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3326\n",
      "Epoch 4: val_loss did not improve from 1.37211\n",
      "31688/31688 [==============================] - 22s 689us/sample - loss: 1.3326 - val_loss: 1.3782\n",
      "Epoch 5/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3295\n",
      "Epoch 5: val_loss improved from 1.37211 to 1.37047, saving model to ./checkpoints/unknown_person_baseline_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.3295 - val_loss: 1.3705\n",
      "Epoch 6/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3280\n",
      "Epoch 6: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 613us/sample - loss: 1.3280 - val_loss: 1.3721\n",
      "Epoch 7/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3264\n",
      "Epoch 7: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 632us/sample - loss: 1.3264 - val_loss: 1.3738\n",
      "Epoch 8/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3366\n",
      "Epoch 8: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 21s 664us/sample - loss: 1.3366 - val_loss: 1.3710\n",
      "Epoch 9/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3352\n",
      "Epoch 9: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 609us/sample - loss: 1.3352 - val_loss: 1.3743\n",
      "Epoch 10/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3366\n",
      "Epoch 10: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 21s 671us/sample - loss: 1.3366 - val_loss: 1.3806\n",
      "Epoch 11/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3353\n",
      "Epoch 11: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3353 - val_loss: 1.3744\n",
      "Epoch 12/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3396\n",
      "Epoch 12: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3396 - val_loss: 1.3743\n",
      "Epoch 13/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3383\n",
      "Epoch 13: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 646us/sample - loss: 1.3383 - val_loss: 1.3778\n",
      "Epoch 14/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3469\n",
      "Epoch 14: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 679us/sample - loss: 1.3469 - val_loss: 1.3710\n",
      "Epoch 15/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3422\n",
      "Epoch 15: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 21s 658us/sample - loss: 1.3422 - val_loss: 1.3784\n",
      "Epoch 16/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3469\n",
      "Epoch 16: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 617us/sample - loss: 1.3469 - val_loss: 1.3861\n",
      "Epoch 17/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3530\n",
      "Epoch 17: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 616us/sample - loss: 1.3530 - val_loss: 1.3770\n",
      "Epoch 18/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3514\n",
      "Epoch 18: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 621us/sample - loss: 1.3514 - val_loss: 1.3840\n",
      "Epoch 19/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3574\n",
      "Epoch 19: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 21s 678us/sample - loss: 1.3574 - val_loss: 1.3763\n",
      "Epoch 20/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3882\n",
      "Epoch 20: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 691us/sample - loss: 1.3882 - val_loss: 1.3877\n",
      "Epoch 21/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3698\n",
      "Epoch 21: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 611us/sample - loss: 1.3698 - val_loss: 1.3924\n",
      "Epoch 22/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4294\n",
      "Epoch 22: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 591us/sample - loss: 1.4294 - val_loss: 1.4183\n",
      "Epoch 23/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4706\n",
      "Epoch 23: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 619us/sample - loss: 1.4706 - val_loss: 1.4276\n",
      "Epoch 24/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.4116\n",
      "Epoch 24: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 627us/sample - loss: 1.4116 - val_loss: 1.3958\n",
      "Epoch 25/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3804\n",
      "Epoch 25: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 689us/sample - loss: 1.3804 - val_loss: 1.3956\n",
      "Epoch 26/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3715\n",
      "Epoch 26: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.3715 - val_loss: 1.3917\n",
      "Epoch 27/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3664\n",
      "Epoch 27: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 692us/sample - loss: 1.3664 - val_loss: 1.3946\n",
      "Epoch 28/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3588\n",
      "Epoch 28: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 696us/sample - loss: 1.3588 - val_loss: 1.3895\n",
      "Epoch 29/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3552\n",
      "Epoch 29: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 609us/sample - loss: 1.3552 - val_loss: 1.3888\n",
      "Epoch 30/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3523\n",
      "Epoch 30: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3523 - val_loss: 1.3814\n",
      "Epoch 31/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3495\n",
      "Epoch 31: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 689us/sample - loss: 1.3495 - val_loss: 1.3813\n",
      "Epoch 32/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3479\n",
      "Epoch 32: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 692us/sample - loss: 1.3479 - val_loss: 1.3768\n",
      "Epoch 33/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3496\n",
      "Epoch 33: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 22s 689us/sample - loss: 1.3496 - val_loss: 1.3817\n",
      "Epoch 34/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3469\n",
      "Epoch 34: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 620us/sample - loss: 1.3469 - val_loss: 1.3775\n",
      "Epoch 35/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3444\n",
      "Epoch 35: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 608us/sample - loss: 1.3444 - val_loss: 1.3779\n",
      "Epoch 36/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3425\n",
      "Epoch 36: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 1.3425 - val_loss: 1.3746\n",
      "Epoch 37/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3435\n",
      "Epoch 37: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 611us/sample - loss: 1.3435 - val_loss: 1.3785\n",
      "Epoch 38/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3422\n",
      "Epoch 38: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 20s 616us/sample - loss: 1.3422 - val_loss: 1.3734\n",
      "Epoch 39/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3401\n",
      "Epoch 39: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 19s 603us/sample - loss: 1.3401 - val_loss: 1.3781\n",
      "Epoch 40/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3397\n",
      "Epoch 40: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 21s 651us/sample - loss: 1.3397 - val_loss: 1.3789\n",
      "Epoch 41/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3356\n",
      "Epoch 41: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 21s 652us/sample - loss: 1.3356 - val_loss: 1.3814\n",
      "Epoch 42/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3355\n",
      "Epoch 42: val_loss did not improve from 1.37047\n",
      "31688/31688 [==============================] - 21s 678us/sample - loss: 1.3355 - val_loss: 1.3756\n",
      "Epoch 43/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3362\n",
      "Epoch 43: val_loss improved from 1.37047 to 1.36807, saving model to ./checkpoints/unknown_person_baseline_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 698us/sample - loss: 1.3362 - val_loss: 1.3681\n",
      "Epoch 44/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3398\n",
      "Epoch 44: val_loss did not improve from 1.36807\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 1.3398 - val_loss: 1.3713\n",
      "Epoch 45/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3346\n",
      "Epoch 45: val_loss did not improve from 1.36807\n",
      "31688/31688 [==============================] - 22s 691us/sample - loss: 1.3346 - val_loss: 1.3700\n",
      "Epoch 46/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3328\n",
      "Epoch 46: val_loss did not improve from 1.36807\n",
      "31688/31688 [==============================] - 22s 691us/sample - loss: 1.3328 - val_loss: 1.3740\n",
      "Epoch 47/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3314\n",
      "Epoch 47: val_loss did not improve from 1.36807\n",
      "31688/31688 [==============================] - 22s 691us/sample - loss: 1.3314 - val_loss: 1.3789\n",
      "Epoch 48/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3315\n",
      "Epoch 48: val_loss did not improve from 1.36807\n",
      "31688/31688 [==============================] - 20s 625us/sample - loss: 1.3315 - val_loss: 1.3717\n",
      "Epoch 49/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3277\n",
      "Epoch 49: val_loss did not improve from 1.36807\n",
      "31688/31688 [==============================] - 19s 613us/sample - loss: 1.3277 - val_loss: 1.3770\n",
      "Epoch 50/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3280\n",
      "Epoch 50: val_loss did not improve from 1.36807\n",
      "31688/31688 [==============================] - 20s 626us/sample - loss: 1.3280 - val_loss: 1.3695\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:41:37.772406: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_6_3/lstm_cell_117/kernel/Assign' id:58358 op device:{requested: '', assigned: ''} def:{{{node lstm_6_3/lstm_cell_117/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_6_3/lstm_cell_117/kernel, lstm_6_3/lstm_cell_117/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:41:41.019794: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_25_3/lstm_cell_136/kernel/m/Assign' id:64385 op device:{requested: '', assigned: ''} def:{{{node lstm_25_3/lstm_cell_136/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_25_3/lstm_cell_136/kernel/m, lstm_25_3/lstm_cell_136/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:41:44.114860: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_3/cond/Merge' id:63349 op device:{requested: '', assigned: ''} def:{{{node dropout_4_3/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_3/cond/Identity, dropout_4_3/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1774)\n",
      "(1514, 1774)\n",
      "(1644, 1774)\n",
      "(1764, 1774)\n",
      "(1836, 1774)\n",
      "(1699, 1774)\n",
      "(1369, 1774)\n",
      "(1778, 1774)\n",
      "(1788, 1774)\n",
      "(1538, 1774)\n",
      "(1908, 1774)\n",
      "(1691, 1774)\n",
      "(1800, 1774)\n",
      "(1848, 1774)\n",
      "(1704, 1774)\n",
      "(1824, 1774)\n",
      "(958, 1774)\n",
      "(1680, 1774)\n",
      "(1860, 1774)\n",
      "{1: 5.543499367898304, 2: 5.253880008390229, 4: 10.0, 5: 5.278606806289648, 6: 5.51121115976016, 8: 8.638978470748526, 9: 6.673295829262623, 10: 7.299430258112917, 12: 8.796504449277855, 13: 6.175746191799874, 17: 9.187092275509414, 19: 8.451135420938156, 21: 8.439750787528794, 22: 1.0, 25: 7.55831927088156, 26: 6.202925430896829, 27: 5.097607462012483, 28: 5.718103997263727, 29: 2.4219039772330007}\n",
      "Train on 31688 samples, validate on 3539 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:45:42.352159: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_7/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0974\n",
      "Epoch 1: val_loss improved from inf to 1.38829, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 28s 877us/sample - loss: 9.0974 - val_loss: 1.3883\n",
      "Epoch 2/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0862\n",
      "Epoch 2: val_loss did not improve from 1.38829\n",
      "31688/31688 [==============================] - 19s 584us/sample - loss: 9.0862 - val_loss: 1.4093\n",
      "Epoch 3/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0264\n",
      "Epoch 3: val_loss did not improve from 1.38829\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 9.0264 - val_loss: 1.3982\n",
      "Epoch 4/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9973\n",
      "Epoch 4: val_loss improved from 1.38829 to 1.38307, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 19s 602us/sample - loss: 8.9973 - val_loss: 1.3831\n",
      "Epoch 5/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9457\n",
      "Epoch 5: val_loss improved from 1.38307 to 1.38049, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 19s 606us/sample - loss: 8.9457 - val_loss: 1.3805\n",
      "Epoch 6/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9208\n",
      "Epoch 6: val_loss improved from 1.38049 to 1.37974, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 19s 590us/sample - loss: 8.9208 - val_loss: 1.3797\n",
      "Epoch 7/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9065\n",
      "Epoch 7: val_loss improved from 1.37974 to 1.37534, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 19s 591us/sample - loss: 8.9065 - val_loss: 1.3753\n",
      "Epoch 8/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8875\n",
      "Epoch 8: val_loss did not improve from 1.37534\n",
      "31688/31688 [==============================] - 20s 630us/sample - loss: 8.8875 - val_loss: 1.3813\n",
      "Epoch 9/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8747\n",
      "Epoch 9: val_loss did not improve from 1.37534\n",
      "31688/31688 [==============================] - 22s 680us/sample - loss: 8.8747 - val_loss: 1.3805\n",
      "Epoch 10/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8570\n",
      "Epoch 10: val_loss improved from 1.37534 to 1.37046, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 8.8570 - val_loss: 1.3705\n",
      "Epoch 11/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8554\n",
      "Epoch 11: val_loss did not improve from 1.37046\n",
      "31688/31688 [==============================] - 22s 682us/sample - loss: 8.8554 - val_loss: 1.3708\n",
      "Epoch 12/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8351\n",
      "Epoch 12: val_loss did not improve from 1.37046\n",
      "31688/31688 [==============================] - 21s 648us/sample - loss: 8.8351 - val_loss: 1.3784\n",
      "Epoch 13/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8158\n",
      "Epoch 13: val_loss did not improve from 1.37046\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 8.8158 - val_loss: 1.3740\n",
      "Epoch 14/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8104\n",
      "Epoch 14: val_loss did not improve from 1.37046\n",
      "31688/31688 [==============================] - 20s 625us/sample - loss: 8.8104 - val_loss: 1.3716\n",
      "Epoch 15/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.7919\n",
      "Epoch 15: val_loss did not improve from 1.37046\n",
      "31688/31688 [==============================] - 20s 647us/sample - loss: 8.7919 - val_loss: 1.3743\n",
      "Epoch 16/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.7821\n",
      "Epoch 16: val_loss did not improve from 1.37046\n",
      "31688/31688 [==============================] - 21s 657us/sample - loss: 8.7821 - val_loss: 1.3716\n",
      "Epoch 17/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.7752\n",
      "Epoch 17: val_loss did not improve from 1.37046\n",
      "31688/31688 [==============================] - 20s 623us/sample - loss: 8.7752 - val_loss: 1.3899\n",
      "Epoch 18/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.7784\n",
      "Epoch 18: val_loss improved from 1.37046 to 1.36792, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 8.7784 - val_loss: 1.3679\n",
      "Epoch 19/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.7616\n",
      "Epoch 19: val_loss did not improve from 1.36792\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 8.7616 - val_loss: 1.3712\n",
      "Epoch 20/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.7661\n",
      "Epoch 20: val_loss improved from 1.36792 to 1.36149, saving model to ./checkpoints/unknown_person_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 8.7661 - val_loss: 1.3615\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:52:39.673147: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_16_4/lstm_cell_164/kernel/Assign' id:79356 op device:{requested: '', assigned: ''} def:{{{node lstm_16_4/lstm_cell_164/kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_16_4/lstm_cell_164/kernel, lstm_16_4/lstm_cell_164/kernel/Initializer/random_uniform)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 15:52:43.958534: W tensorflow/c/c_api.cc:304] Operation '{name:'dense_1_4/bias/v/Assign' id:84625 op device:{requested: '', assigned: ''} def:{{{node dense_1_4/bias/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](dense_1_4/bias/v, dense_1_4/bias/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31688 samples, validate on 3539 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:52:50.806626: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:53:13.066496: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_8/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3327"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 15:53:36.623598: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_9/mul' id:83337 op device:{requested: '', assigned: ''} def:{{{node loss_9/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_9/mul/x, loss_9/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.37097, saving model to ./checkpoints/unknown_person_baseline_2_p11_21.h5\n",
      "31688/31688 [==============================] - 33s 1ms/sample - loss: 1.3327 - val_loss: 1.3710\n",
      "Epoch 2/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3320\n",
      "Epoch 2: val_loss did not improve from 1.37097\n",
      "31688/31688 [==============================] - 22s 694us/sample - loss: 1.3320 - val_loss: 1.3742\n",
      "Epoch 3/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3273\n",
      "Epoch 3: val_loss did not improve from 1.37097\n",
      "31688/31688 [==============================] - 22s 692us/sample - loss: 1.3273 - val_loss: 1.3785\n",
      "Epoch 4/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3280\n",
      "Epoch 4: val_loss did not improve from 1.37097\n",
      "31688/31688 [==============================] - 22s 691us/sample - loss: 1.3280 - val_loss: 1.3740\n",
      "Epoch 5/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3265\n",
      "Epoch 5: val_loss improved from 1.37097 to 1.36726, saving model to ./checkpoints/unknown_person_baseline_2_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 697us/sample - loss: 1.3265 - val_loss: 1.3673\n",
      "Epoch 6/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3267\n",
      "Epoch 6: val_loss did not improve from 1.36726\n",
      "31688/31688 [==============================] - 22s 680us/sample - loss: 1.3267 - val_loss: 1.3703\n",
      "Epoch 7/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3252\n",
      "Epoch 7: val_loss did not improve from 1.36726\n",
      "31688/31688 [==============================] - 22s 687us/sample - loss: 1.3252 - val_loss: 1.3710\n",
      "Epoch 8/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3220\n",
      "Epoch 8: val_loss improved from 1.36726 to 1.36527, saving model to ./checkpoints/unknown_person_baseline_2_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 680us/sample - loss: 1.3220 - val_loss: 1.3653\n",
      "Epoch 9/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3196\n",
      "Epoch 9: val_loss improved from 1.36527 to 1.36467, saving model to ./checkpoints/unknown_person_baseline_2_p11_21.h5\n",
      "31688/31688 [==============================] - 22s 684us/sample - loss: 1.3196 - val_loss: 1.3647\n",
      "Epoch 10/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3192\n",
      "Epoch 10: val_loss improved from 1.36467 to 1.36295, saving model to ./checkpoints/unknown_person_baseline_2_p11_21.h5\n",
      "31688/31688 [==============================] - 20s 630us/sample - loss: 1.3192 - val_loss: 1.3629\n",
      "Epoch 11/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3201\n",
      "Epoch 11: val_loss did not improve from 1.36295\n",
      "31688/31688 [==============================] - 19s 603us/sample - loss: 1.3201 - val_loss: 1.3651\n",
      "Epoch 12/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3185\n",
      "Epoch 12: val_loss did not improve from 1.36295\n",
      "31688/31688 [==============================] - 21s 674us/sample - loss: 1.3185 - val_loss: 1.3649\n",
      "Epoch 13/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3180\n",
      "Epoch 13: val_loss improved from 1.36295 to 1.36056, saving model to ./checkpoints/unknown_person_baseline_2_p11_21.h5\n",
      "31688/31688 [==============================] - 20s 641us/sample - loss: 1.3180 - val_loss: 1.3606\n",
      "Epoch 14/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3142\n",
      "Epoch 14: val_loss did not improve from 1.36056\n",
      "31688/31688 [==============================] - 21s 674us/sample - loss: 1.3142 - val_loss: 1.3612\n",
      "Epoch 15/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3158\n",
      "Epoch 15: val_loss did not improve from 1.36056\n",
      "31688/31688 [==============================] - 21s 668us/sample - loss: 1.3158 - val_loss: 1.3699\n",
      "Epoch 16/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3119\n",
      "Epoch 16: val_loss did not improve from 1.36056\n",
      "31688/31688 [==============================] - 19s 611us/sample - loss: 1.3119 - val_loss: 1.3648\n",
      "Epoch 17/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3137\n",
      "Epoch 17: val_loss did not improve from 1.36056\n",
      "31688/31688 [==============================] - 21s 661us/sample - loss: 1.3137 - val_loss: 1.3684\n",
      "Epoch 18/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3133\n",
      "Epoch 18: val_loss did not improve from 1.36056\n",
      "31688/31688 [==============================] - 21s 660us/sample - loss: 1.3133 - val_loss: 1.3754\n",
      "Epoch 19/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3106\n",
      "Epoch 19: val_loss did not improve from 1.36056\n",
      "31688/31688 [==============================] - 20s 626us/sample - loss: 1.3106 - val_loss: 1.3645\n",
      "Epoch 20/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3077\n",
      "Epoch 20: val_loss improved from 1.36056 to 1.35634, saving model to ./checkpoints/unknown_person_baseline_2_p11_21.h5\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 1.3077 - val_loss: 1.3563\n",
      "Train on 31688 samples, validate on 3539 samples\n",
      "Epoch 1/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3091\n",
      "Epoch 1: val_loss improved from inf to 1.35901, saving model to ./checkpoints/unknown_person_baseline_p11_22.h5\n",
      "31688/31688 [==============================] - 20s 628us/sample - loss: 1.3091 - val_loss: 1.3590\n",
      "Epoch 2/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3095\n",
      "Epoch 2: val_loss did not improve from 1.35901\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3095 - val_loss: 1.3612\n",
      "Epoch 3/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3057\n",
      "Epoch 3: val_loss improved from 1.35901 to 1.35238, saving model to ./checkpoints/unknown_person_baseline_p11_22.h5\n",
      "31688/31688 [==============================] - 21s 671us/sample - loss: 1.3057 - val_loss: 1.3524\n",
      "Epoch 4/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3068\n",
      "Epoch 4: val_loss improved from 1.35238 to 1.35210, saving model to ./checkpoints/unknown_person_baseline_p11_22.h5\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3068 - val_loss: 1.3521\n",
      "Epoch 5/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3045\n",
      "Epoch 5: val_loss did not improve from 1.35210\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3045 - val_loss: 1.3554\n",
      "Epoch 6/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3017\n",
      "Epoch 6: val_loss did not improve from 1.35210\n",
      "31688/31688 [==============================] - 21s 659us/sample - loss: 1.3017 - val_loss: 1.3626\n",
      "Epoch 7/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3014\n",
      "Epoch 7: val_loss did not improve from 1.35210\n",
      "31688/31688 [==============================] - 22s 683us/sample - loss: 1.3014 - val_loss: 1.3594\n",
      "Epoch 8/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3069\n",
      "Epoch 8: val_loss improved from 1.35210 to 1.35034, saving model to ./checkpoints/unknown_person_baseline_p11_22.h5\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3069 - val_loss: 1.3503\n",
      "Epoch 9/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3116\n",
      "Epoch 9: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 660us/sample - loss: 1.3116 - val_loss: 1.3522\n",
      "Epoch 10/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3096\n",
      "Epoch 10: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 646us/sample - loss: 1.3096 - val_loss: 1.3666\n",
      "Epoch 11/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3160\n",
      "Epoch 11: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 604us/sample - loss: 1.3160 - val_loss: 1.3598\n",
      "Epoch 12/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3547\n",
      "Epoch 12: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 624us/sample - loss: 1.3547 - val_loss: 1.3684\n",
      "Epoch 13/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3242\n",
      "Epoch 13: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 598us/sample - loss: 1.3242 - val_loss: 1.3662\n",
      "Epoch 14/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3240\n",
      "Epoch 14: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 658us/sample - loss: 1.3240 - val_loss: 1.3642\n",
      "Epoch 15/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3246\n",
      "Epoch 15: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 691us/sample - loss: 1.3246 - val_loss: 1.3694\n",
      "Epoch 16/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3403\n",
      "Epoch 16: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 658us/sample - loss: 1.3403 - val_loss: 1.3724\n",
      "Epoch 17/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3409\n",
      "Epoch 17: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 668us/sample - loss: 1.3409 - val_loss: 1.3790\n",
      "Epoch 18/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3358\n",
      "Epoch 18: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3358 - val_loss: 1.3680\n",
      "Epoch 19/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3454\n",
      "Epoch 19: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 667us/sample - loss: 1.3454 - val_loss: 1.3692\n",
      "Epoch 20/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3570\n",
      "Epoch 20: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 658us/sample - loss: 1.3570 - val_loss: 1.3755\n",
      "Epoch 21/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3567\n",
      "Epoch 21: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 630us/sample - loss: 1.3567 - val_loss: 1.3792\n",
      "Epoch 22/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3943\n",
      "Epoch 22: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 622us/sample - loss: 1.3943 - val_loss: 1.3868\n",
      "Epoch 23/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3962\n",
      "Epoch 23: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 613us/sample - loss: 1.3962 - val_loss: 1.3939\n",
      "Epoch 24/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3808\n",
      "Epoch 24: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 679us/sample - loss: 1.3808 - val_loss: 1.3962\n",
      "Epoch 25/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3754\n",
      "Epoch 25: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 671us/sample - loss: 1.3754 - val_loss: 1.3878\n",
      "Epoch 26/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3691\n",
      "Epoch 26: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 637us/sample - loss: 1.3691 - val_loss: 1.3947\n",
      "Epoch 27/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3614\n",
      "Epoch 27: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 612us/sample - loss: 1.3614 - val_loss: 1.3863\n",
      "Epoch 28/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3609\n",
      "Epoch 28: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 664us/sample - loss: 1.3609 - val_loss: 1.3871\n",
      "Epoch 29/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3626\n",
      "Epoch 29: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 658us/sample - loss: 1.3626 - val_loss: 1.3788\n",
      "Epoch 30/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3650\n",
      "Epoch 30: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 692us/sample - loss: 1.3650 - val_loss: 1.3844\n",
      "Epoch 31/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3604\n",
      "Epoch 31: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 630us/sample - loss: 1.3604 - val_loss: 1.3855\n",
      "Epoch 32/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3579\n",
      "Epoch 32: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 609us/sample - loss: 1.3579 - val_loss: 1.3828\n",
      "Epoch 33/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3583\n",
      "Epoch 33: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 611us/sample - loss: 1.3583 - val_loss: 1.3837\n",
      "Epoch 34/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3519\n",
      "Epoch 34: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 635us/sample - loss: 1.3519 - val_loss: 1.3793\n",
      "Epoch 35/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3537\n",
      "Epoch 35: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 612us/sample - loss: 1.3537 - val_loss: 1.3775\n",
      "Epoch 36/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3534\n",
      "Epoch 36: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 607us/sample - loss: 1.3534 - val_loss: 1.3749\n",
      "Epoch 37/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3518\n",
      "Epoch 37: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 611us/sample - loss: 1.3518 - val_loss: 1.3847\n",
      "Epoch 38/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3454\n",
      "Epoch 38: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 614us/sample - loss: 1.3454 - val_loss: 1.3794\n",
      "Epoch 39/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3448\n",
      "Epoch 39: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 673us/sample - loss: 1.3448 - val_loss: 1.3781\n",
      "Epoch 40/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3462\n",
      "Epoch 40: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 686us/sample - loss: 1.3462 - val_loss: 1.3864\n",
      "Epoch 41/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3484\n",
      "Epoch 41: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.3484 - val_loss: 1.3801\n",
      "Epoch 42/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3453\n",
      "Epoch 42: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 687us/sample - loss: 1.3453 - val_loss: 1.3782\n",
      "Epoch 43/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3430\n",
      "Epoch 43: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3430 - val_loss: 1.3763\n",
      "Epoch 44/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3425\n",
      "Epoch 44: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3425 - val_loss: 1.3774\n",
      "Epoch 45/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3406\n",
      "Epoch 45: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 668us/sample - loss: 1.3406 - val_loss: 1.3749\n",
      "Epoch 46/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3420\n",
      "Epoch 46: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 687us/sample - loss: 1.3420 - val_loss: 1.3761\n",
      "Epoch 47/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3370\n",
      "Epoch 47: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 22s 685us/sample - loss: 1.3370 - val_loss: 1.3746\n",
      "Epoch 48/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3367\n",
      "Epoch 48: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 20s 618us/sample - loss: 1.3367 - val_loss: 1.3704\n",
      "Epoch 49/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3504\n",
      "Epoch 49: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 19s 613us/sample - loss: 1.3504 - val_loss: 1.3731\n",
      "Epoch 50/50\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3455\n",
      "Epoch 50: val_loss did not improve from 1.35034\n",
      "31688/31688 [==============================] - 21s 660us/sample - loss: 1.3455 - val_loss: 1.3751\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:17:42.309067: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_19_5/lstm_cell_204/bias/Assign' id:99225 op device:{requested: '', assigned: ''} def:{{{node lstm_19_5/lstm_cell_204/bias/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_19_5/lstm_cell_204/bias, lstm_19_5/lstm_cell_204/bias/Initializer/concat)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:17:47.493036: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_31_5/lstm_cell_216/kernel/v/Assign' id:103876 op device:{requested: '', assigned: ''} def:{{{node lstm_31_5/lstm_cell_216/kernel/v/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_31_5/lstm_cell_216/kernel/v, lstm_31_5/lstm_cell_216/kernel/v/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:17:52.493971: W tensorflow/c/c_api.cc:304] Operation '{name:'dropout_4_5/cond/Merge' id:102107 op device:{requested: '', assigned: ''} def:{{{node dropout_4_5/cond/Merge}} = Merge[N=2, T=DT_FLOAT, _has_manual_control_dependencies=true](dropout_4_5/cond/Identity, dropout_4_5/cond/dropout/SelectV2)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(1485, 1774)\n",
      "(1514, 1774)\n",
      "(1644, 1774)\n",
      "(1764, 1774)\n",
      "(1836, 1774)\n",
      "(1699, 1774)\n",
      "(1369, 1774)\n",
      "(1778, 1774)\n",
      "(1788, 1774)\n",
      "(1538, 1774)\n",
      "(1908, 1774)\n",
      "(1691, 1774)\n",
      "(1800, 1774)\n",
      "(1848, 1774)\n",
      "(1704, 1774)\n",
      "(1824, 1774)\n",
      "(958, 1774)\n",
      "(1680, 1774)\n",
      "(1860, 1774)\n",
      "{1: 5.710981038395298, 2: 5.269134309296075, 4: 10.0, 5: 5.806062180697036, 6: 5.710973995217291, 8: 8.730995402565556, 9: 6.588936067312435, 10: 7.444451531098859, 12: 8.813372412536935, 13: 6.414156083476528, 17: 9.141836301282895, 19: 8.492573262257233, 21: 8.543696169821978, 22: 1.0, 25: 7.497874036282932, 26: 6.122746354226073, 27: 4.7974966980212, 28: 5.661997496150218, 29: 3.168340953986722}\n",
      "Train on 31688 samples, validate on 3539 samples\n",
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:22:34.319506: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_9/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31688/31688 [==============================] - ETA: 0s - loss: 9.1271\n",
      "Epoch 1: val_loss improved from inf to 1.37590, saving model to ./checkpoints/unknown_person_p11_22.h5\n",
      "31688/31688 [==============================] - 33s 1ms/sample - loss: 9.1271 - val_loss: 1.3759\n",
      "Epoch 2/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0776\n",
      "Epoch 2: val_loss improved from 1.37590 to 1.37455, saving model to ./checkpoints/unknown_person_p11_22.h5\n",
      "31688/31688 [==============================] - 20s 623us/sample - loss: 9.0776 - val_loss: 1.3745\n",
      "Epoch 3/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0622\n",
      "Epoch 3: val_loss improved from 1.37455 to 1.36338, saving model to ./checkpoints/unknown_person_p11_22.h5\n",
      "31688/31688 [==============================] - 19s 608us/sample - loss: 9.0622 - val_loss: 1.3634\n",
      "Epoch 4/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 9.0102\n",
      "Epoch 4: val_loss did not improve from 1.36338\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 9.0102 - val_loss: 1.3716\n",
      "Epoch 5/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9787\n",
      "Epoch 5: val_loss did not improve from 1.36338\n",
      "31688/31688 [==============================] - 22s 697us/sample - loss: 8.9787 - val_loss: 1.3711\n",
      "Epoch 6/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9379\n",
      "Epoch 6: val_loss did not improve from 1.36338\n",
      "31688/31688 [==============================] - 20s 634us/sample - loss: 8.9379 - val_loss: 1.3686\n",
      "Epoch 7/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9193\n",
      "Epoch 7: val_loss improved from 1.36338 to 1.36239, saving model to ./checkpoints/unknown_person_p11_22.h5\n",
      "31688/31688 [==============================] - 22s 686us/sample - loss: 8.9193 - val_loss: 1.3624\n",
      "Epoch 8/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8840\n",
      "Epoch 8: val_loss did not improve from 1.36239\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 8.8840 - val_loss: 1.3713\n",
      "Epoch 9/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.9118\n",
      "Epoch 9: val_loss did not improve from 1.36239\n",
      "31688/31688 [==============================] - 21s 671us/sample - loss: 8.9118 - val_loss: 1.3793\n",
      "Epoch 10/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8847\n",
      "Epoch 10: val_loss improved from 1.36239 to 1.35873, saving model to ./checkpoints/unknown_person_p11_22.h5\n",
      "31688/31688 [==============================] - 22s 686us/sample - loss: 8.8847 - val_loss: 1.3587\n",
      "Epoch 11/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8627\n",
      "Epoch 11: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 19s 603us/sample - loss: 8.8627 - val_loss: 1.3626\n",
      "Epoch 12/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8531\n",
      "Epoch 12: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 21s 664us/sample - loss: 8.8531 - val_loss: 1.3625\n",
      "Epoch 13/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8591\n",
      "Epoch 13: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 8.8591 - val_loss: 1.3617\n",
      "Epoch 14/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8478\n",
      "Epoch 14: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 22s 693us/sample - loss: 8.8478 - val_loss: 1.3664\n",
      "Epoch 15/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8482\n",
      "Epoch 15: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 22s 680us/sample - loss: 8.8482 - val_loss: 1.3656\n",
      "Epoch 16/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8228\n",
      "Epoch 16: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 21s 666us/sample - loss: 8.8228 - val_loss: 1.3709\n",
      "Epoch 17/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8111\n",
      "Epoch 17: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 20s 636us/sample - loss: 8.8111 - val_loss: 1.3606\n",
      "Epoch 18/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8330\n",
      "Epoch 18: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 19s 603us/sample - loss: 8.8330 - val_loss: 1.3606\n",
      "Epoch 19/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.7962\n",
      "Epoch 19: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 19s 604us/sample - loss: 8.7962 - val_loss: 1.3646\n",
      "Epoch 20/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 8.8110\n",
      "Epoch 20: val_loss did not improve from 1.35873\n",
      "31688/31688 [==============================] - 20s 621us/sample - loss: 8.8110 - val_loss: 1.3758\n",
      "WARNING:tensorflow:Layer lstm will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_1 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_2 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_3 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_4 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_5 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_6 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_7 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_8 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_9 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_10 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_11 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_12 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_13 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_14 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_15 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_16 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_17 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_18 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_19 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_20 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_21 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_22 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_23 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_24 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_25 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_26 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_27 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_28 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_29 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_30 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_31 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_32 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_33 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_34 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_35 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n",
      "WARNING:tensorflow:Layer lstm_36 will not use cuDNN kernels since it doesn't meet the criteria. It will use a generic GPU kernel as fallback when running on GPU.\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:29:45.411028: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_36_6/lstm_cell_258/recurrent_kernel/Assign' id:121336 op device:{requested: '', assigned: ''} def:{{{node lstm_36_6/lstm_cell_258/recurrent_kernel/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_36_6/lstm_cell_258/recurrent_kernel, lstm_36_6/lstm_cell_258/recurrent_kernel/Initializer/mul_1)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n",
      "2023-11-08 16:29:51.516719: W tensorflow/c/c_api.cc:304] Operation '{name:'lstm_30_6/lstm_cell_252/kernel/m/Assign' id:122616 op device:{requested: '', assigned: ''} def:{{{node lstm_30_6/lstm_cell_252/kernel/m/Assign}} = AssignVariableOp[_has_manual_control_dependencies=true, dtype=DT_FLOAT, validate_shape=false](lstm_30_6/lstm_cell_252/kernel/m, lstm_30_6/lstm_cell_252/kernel/m/Initializer/zeros)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Train on 31688 samples, validate on 3539 samples\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:30:00.622689: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Epoch 1/20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:30:33.801510: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape indropout_10/cond/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3104"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2023-11-08 16:30:56.565019: W tensorflow/c/c_api.cc:304] Operation '{name:'loss_13/mul' id:122095 op device:{requested: '', assigned: ''} def:{{{node loss_13/mul}} = Mul[T=DT_FLOAT, _has_manual_control_dependencies=true](loss_13/mul/x, loss_13/dense_3_loss/value)}}' was changed by setting attribute after it was run by a session. This mutation will have no effect, and will trigger an error in the future. Either don't modify nodes after running them or create a new session.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Epoch 1: val_loss improved from inf to 1.35275, saving model to ./checkpoints/unknown_person_baseline_2_p11_22.h5\n",
      "31688/31688 [==============================] - 36s 1ms/sample - loss: 1.3104 - val_loss: 1.3527\n",
      "Epoch 2/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3045\n",
      "Epoch 2: val_loss improved from 1.35275 to 1.35159, saving model to ./checkpoints/unknown_person_baseline_2_p11_22.h5\n",
      "31688/31688 [==============================] - 21s 664us/sample - loss: 1.3045 - val_loss: 1.3516\n",
      "Epoch 3/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3120\n",
      "Epoch 3: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 21s 657us/sample - loss: 1.3120 - val_loss: 1.3614\n",
      "Epoch 4/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3052\n",
      "Epoch 4: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 21s 664us/sample - loss: 1.3052 - val_loss: 1.3628\n",
      "Epoch 5/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3065\n",
      "Epoch 5: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 21s 655us/sample - loss: 1.3065 - val_loss: 1.3544\n",
      "Epoch 6/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3020\n",
      "Epoch 6: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3020 - val_loss: 1.3573\n",
      "Epoch 7/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3016\n",
      "Epoch 7: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 22s 688us/sample - loss: 1.3016 - val_loss: 1.3604\n",
      "Epoch 8/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2983\n",
      "Epoch 8: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 20s 638us/sample - loss: 1.2983 - val_loss: 1.3550\n",
      "Epoch 9/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.3003\n",
      "Epoch 9: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 19s 597us/sample - loss: 1.3003 - val_loss: 1.3606\n",
      "Epoch 10/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2961\n",
      "Epoch 10: val_loss did not improve from 1.35159\n",
      "31688/31688 [==============================] - 19s 609us/sample - loss: 1.2961 - val_loss: 1.3543\n",
      "Epoch 11/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2947\n",
      "Epoch 11: val_loss improved from 1.35159 to 1.34569, saving model to ./checkpoints/unknown_person_baseline_2_p11_22.h5\n",
      "31688/31688 [==============================] - 19s 598us/sample - loss: 1.2947 - val_loss: 1.3457\n",
      "Epoch 12/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2955\n",
      "Epoch 12: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 19s 598us/sample - loss: 1.2955 - val_loss: 1.3569\n",
      "Epoch 13/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2946\n",
      "Epoch 13: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 21s 672us/sample - loss: 1.2946 - val_loss: 1.3581\n",
      "Epoch 14/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2973\n",
      "Epoch 14: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 20s 638us/sample - loss: 1.2973 - val_loss: 1.3559\n",
      "Epoch 15/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2906\n",
      "Epoch 15: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 20s 641us/sample - loss: 1.2906 - val_loss: 1.3579\n",
      "Epoch 16/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2910\n",
      "Epoch 16: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 22s 690us/sample - loss: 1.2910 - val_loss: 1.3529\n",
      "Epoch 17/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2899\n",
      "Epoch 17: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 22s 694us/sample - loss: 1.2899 - val_loss: 1.3497\n",
      "Epoch 18/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2915\n",
      "Epoch 18: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 22s 689us/sample - loss: 1.2915 - val_loss: 1.3523\n",
      "Epoch 19/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2882\n",
      "Epoch 19: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 20s 634us/sample - loss: 1.2882 - val_loss: 1.3549\n",
      "Epoch 20/20\n",
      "31688/31688 [==============================] - ETA: 0s - loss: 1.2891\n",
      "Epoch 20: val_loss did not improve from 1.34569\n",
      "31688/31688 [==============================] - 20s 623us/sample - loss: 1.2891 - val_loss: 1.3530\n"
     ]
    }
   ],
   "source": [
    "person_nums = [1,2,4,5,6,8,9,10,11,12,13,17,19,21,22,25,26,27,28,29]\n",
    "## Build the baseline model for emotion recognition with dropout layers\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, SimpleRNN, LSTM, Conv2D, Flatten, MaxPooling2D, GRU, AveragePooling2D, Dropout, Lambda\n",
    "from tensorflow.keras.callbacks import ModelCheckpoint\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "def individual_model(features_list):\n",
    "    input_layers = []\n",
    "    hidden_layers = []\n",
    "    combined_feature = np.empty((len(features_list[0]),0))\n",
    "    for i, feature in enumerate(features_list):\n",
    "        \n",
    "        if len(feature.shape) == 3:\n",
    "            input_i = Input(shape=(feature.shape[1], feature.shape[2]))\n",
    "            input_layers.append(input_i)\n",
    "\n",
    "            hidden_i = input_i[:,:,:,None]\n",
    "            hidden_i = Conv2D(32, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(16, 3, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((3, 3))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(8, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Conv2D(4, 1, activation='relu')(hidden_i)\n",
    "            hidden_i = AveragePooling2D((2, 2))(hidden_i)\n",
    "            hidden_i = Dropout(0.5)(hidden_i)\n",
    "            hidden_i = Flatten()(hidden_i)\n",
    "\n",
    "            hidden_layers.append(hidden_i)\n",
    "        elif feature.shape[1] <20:\n",
    "            combined_feature = np.hstack((combined_feature, feature))\n",
    "            \n",
    "        else:  # For series features\n",
    "            input_i = Input(shape=(feature.shape[1],))\n",
    "            input_layers.append(input_i)\n",
    "            hidden_i = Lambda(lambda x: x[:, :, None])(input_i)  # Add a new dimension\n",
    "            hidden_i = LSTM(4)(hidden_i)\n",
    "            hidden_layers.append(hidden_i)\n",
    "    input_i = Input(shape=(combined_feature.shape[1],))\n",
    "    input_layers.append(input_i)\n",
    "    dense_num = np.max((1, int(combined_feature.shape[1]/2)))\n",
    "    hidden_i = Dense(dense_num, activation='relu')(input_i)\n",
    "    hidden_layers.append(hidden_i)\n",
    "    print(combined_feature.shape)\n",
    "    concat_layer = concatenate(hidden_layers)\n",
    "    h = Dropout(0.2)(concat_layer)\n",
    "    h = Dense(64, activation='relu')(h)\n",
    "    h = Dense(32, activation='relu')(h)\n",
    "    output_layer = Dense(2)(h)\n",
    "    model = Model(inputs=input_layers, outputs=output_layer)\n",
    "    model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "    return model\n",
    "\n",
    "import tensorflow as tf\n",
    "from tensorflow.keras.callbacks import Callback\n",
    "import numpy as np\n",
    "\n",
    "class PruningCallback(tf.keras.callbacks.Callback):\n",
    "    def __init__(self, initial_pruning_factor=0.1, final_pruning_factor=0.5, start_epoch=0, end_epoch=None, frequency=1):\n",
    "        super(PruningCallback, self).__init__()\n",
    "        self.initial_pruning_factor = initial_pruning_factor\n",
    "        self.final_pruning_factor = final_pruning_factor\n",
    "        self.start_epoch = start_epoch\n",
    "        self.end_epoch = end_epoch if end_epoch is not None else np.inf\n",
    "        self.frequency = frequency\n",
    "        self.pruned_weights = {}\n",
    "        self.layer_importance = {}\n",
    "\n",
    "    def get_pruning_factor(self, epoch):\n",
    "        if epoch < self.start_epoch:\n",
    "            return 0\n",
    "        if epoch > self.end_epoch:\n",
    "            return self.final_pruning_factor\n",
    "        return self.initial_pruning_factor + (self.final_pruning_factor - self.initial_pruning_factor) * (epoch - self.start_epoch) / (self.end_epoch - self.start_epoch)\n",
    "\n",
    "    def on_train_begin(self, logs=None):\n",
    "        total_weight_magnitude = 0\n",
    "        for layer in self.model.layers:\n",
    "            if hasattr(layer, 'get_weights'):\n",
    "                weights = layer.get_weights()\n",
    "                layer_norm = sum(np.linalg.norm(w) for w in weights)\n",
    "                total_weight_magnitude += layer_norm\n",
    "                self.layer_importance[layer.name] = layer_norm\n",
    "    \n",
    "        # Normalize the layer importance values so they sum up to 1\n",
    "        for layer_name in self.layer_importance:\n",
    "            self.layer_importance[layer_name] /= total_weight_magnitude\n",
    "    # def on_train_begin(self, logs=None):\n",
    "    #     total_weight_magnitude = sum([np.linalg.norm(layer.get_weights()) for layer in self.model.layers if hasattr(layer, 'get_weights')])\n",
    "    #     for layer in self.model.layers:\n",
    "    #         if hasattr(layer, 'get_weights'):\n",
    "    #             self.layer_importance[layer.name] = np.linalg.norm(layer.get_weights()) / total_weight_magnitude\n",
    "\n",
    "    def get_layer_pruning_factor(self, layer_name, global_pruning_factor):\n",
    "        if layer_name in self.layer_importance:\n",
    "            importance = self.layer_importance[layer_name]\n",
    "            adjusted_pruning_factor = global_pruning_factor * (1 - importance)\n",
    "            return min(adjusted_pruning_factor, 1)  # Ensure the pruning factor is not greater than 1\n",
    "        return global_pruning_factor\n",
    "    def prune_weights(self, layer, global_pruning_factor):\n",
    "        \n",
    "        weights = layer.get_weights()\n",
    "        layer_name = layer.name\n",
    "        pruning_factor = self.get_layer_pruning_factor(layer_name, global_pruning_factor)\n",
    "\n",
    "        if layer_name not in self.pruned_weights:\n",
    "            self.pruned_weights[layer_name] = [np.zeros_like(w, dtype=bool) for w in weights]\n",
    "        \n",
    "        for i in range(len(weights)):\n",
    "            weight = weights[i]\n",
    "            # print(weight.shape)\n",
    "            # print(weight.size)\n",
    "            if weight.ndim > 1:  # Only prune dense or convolutional layers\n",
    "                unpruned_weights = np.logical_not(self.pruned_weights[layer_name][i])\n",
    "                num_unpruned = np.sum(unpruned_weights)\n",
    "                num_pruning = min(num_unpruned, int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i]))\n",
    "                num_pruning = int(weight.size * pruning_factor) - np.sum(self.pruned_weights[layer_name][i])\n",
    "                if num_pruning > 0:\n",
    "                    unpruned_flat_indices = np.flatnonzero(unpruned_weights)\n",
    "                    abs_unpruned_weights = np.abs(weight[unpruned_weights])\n",
    "                    pruning_flat_indices = np.argpartition(abs_unpruned_weights, num_pruning)[:num_pruning]\n",
    "                    \n",
    "                    indices = np.unravel_index(pruning_flat_indices, weight.shape)\n",
    "                    self.pruned_weights[layer_name][i][indices] = True\n",
    "\n",
    "                weights[i] = weights[i]*(~self.pruned_weights[layer_name][i])\n",
    "                \n",
    "        layer.set_weights(weights)\n",
    "\n",
    "    def on_epoch_end(self, epoch, logs=None):\n",
    "        if (epoch - self.start_epoch) % self.frequency != 0:\n",
    "            return\n",
    "\n",
    "        pruning_factor = self.get_pruning_factor(epoch)\n",
    "        for layer in self.model.layers:\n",
    "            if isinstance(layer, (tf.keras.layers.Dense, tf.keras.layers.Conv2D)):\n",
    "                self.prune_weights(layer, pruning_factor)\n",
    "from feature_emotion import feature_extract, split_data, label_unique_tuples, split_data_unknown\n",
    "gts, sensor_nums, walk_nums, trace_nums, people_nums, spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features = feature_extract(person_nums)\n",
    "\n",
    "walk_nums_all = np.squeeze(walk_nums)\n",
    "trace_nums_all = np.squeeze(trace_nums)\n",
    "people_nums_all = np.squeeze(people_nums)\n",
    "\n",
    "## 0: train, 1: validation 2: test\n",
    "\n",
    "test_person_id = [11]\n",
    "flag_tr_val_te = split_data_unknown(test_person_id, walk_nums_all, trace_nums_all, people_nums_all, person_nums, rand_seed = 42)\n",
    "## Data Normalization before training ans testing\n",
    "import tensorflow as tf\n",
    "tf.compat.v1.disable_v2_behavior()\n",
    "from tensorflow.keras.models import Model\n",
    "from tensorflow.keras.layers import Input, Dense, concatenate, LSTM\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "scalers = []\n",
    "X_train_normalized = []\n",
    "X_val_normalized = []\n",
    "X_test_normalized = []\n",
    "train_idx = np.where(flag_tr_val_te ==0)[0]\n",
    "np.random.shuffle(train_idx)\n",
    "val_idx = np.where(flag_tr_val_te ==1)[0]\n",
    "test_idx = np.where(flag_tr_val_te ==2)[0]\n",
    "\n",
    "for i, feature in enumerate([spe_centr, delta_spe_centr, spe_crest, delta_spe_crest, spe_decrease, delta_spe_decrease, spe_entropy, delta_spe_entropy, spe_flatness, delta_spe_flatness, spe_flux, delta_spe_flux, spe_kurtosis, delta_spe_kurtosis, spe_skewness, delta_spe_skewness, spe_rfp, delta_spe_rfp, spe_slope, delta_spe_slope, spe_spread, delta_spe_spread, wlk_fre, wlk_fres_trace, cwt_figs_all, cwt_sum_all_0, cwt_sum_all_1, cwt_sum_all_2, cwt_sum_all_3, high_fre_compos, pitchs, low_fre_compos, auto_corrs, real_hils, imag_hils, dur_time_1_alls, dur_time_2_alls, jitters, shimmers, jitter_rap, hrs, feature, slope, zcrs, fft_features, energy_alls, log_energy_alls, smoothe_energy_alls, legendres, double_support_time, pdps_new, lpcs, ceps_features]):\n",
    "    scaler = StandardScaler()\n",
    "    if len(feature.shape)==2:\n",
    "        X_train_i = feature[train_idx,:]\n",
    "        X_val_i = feature[val_idx,:]\n",
    "        X_test_i = feature[test_idx,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i)\n",
    "        scalers.append(scaler)\n",
    "    else:\n",
    "        X_train_i = feature[train_idx,:,:]\n",
    "        X_val_i = feature[val_idx,:,:]\n",
    "        X_test_i = feature[test_idx,:,:]\n",
    "        X_train_normalized_i = scaler.fit_transform(X_train_i.reshape(X_train_i.shape[0], -1)).reshape(X_train_i.shape)\n",
    "        X_val_normalized_i = scaler.transform(X_val_i.reshape(X_val_i.shape[0], -1)).reshape(X_val_i.shape)\n",
    "        X_test_normalized_i = scaler.transform(X_test_i.reshape(X_test_i.shape[0], -1)).reshape(X_test_i.shape)\n",
    "        scalers.append(scaler)\n",
    "    X_train_normalized.append(X_train_normalized_i)\n",
    "    X_val_normalized.append(X_val_normalized_i)\n",
    "    X_test_normalized.append(X_test_normalized_i)\n",
    "y_train = gts[train_idx,:]\n",
    "y_val = gts[val_idx,:]\n",
    "y_test = gts[test_idx,:]\n",
    "X_train_normalized_new = []\n",
    "combined_feature = np.empty((len(X_train_normalized[0]),0))\n",
    "for feature in X_train_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_train_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_train_normalized_new.append(feature)\n",
    "X_train_normalized_new.append(combined_feature)\n",
    "\n",
    "X_val_normalized_new = []\n",
    "combined_feature = np.empty((len(X_val_normalized[0]),0))\n",
    "for feature in X_val_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_val_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_val_normalized_new.append(feature)\n",
    "X_val_normalized_new.append(combined_feature)\n",
    "\n",
    "X_test_normalized_new = []\n",
    "combined_feature = np.empty((len(X_test_normalized[0]),0))\n",
    "for feature in X_test_normalized:\n",
    "    if len(feature.shape) == 3:\n",
    "        X_test_normalized_new.append(feature)\n",
    "    elif feature.shape[1] <20:\n",
    "        combined_feature = np.hstack((combined_feature, feature))\n",
    "    else:\n",
    "        X_test_normalized_new.append(feature)\n",
    "X_test_normalized_new.append(combined_feature)\n",
    "\n",
    "model = individual_model(X_train_normalized)\n",
    "model.compile(loss='mean_absolute_error', optimizer='adam')\n",
    "num_epochs = 50\n",
    "\n",
    "\n",
    "# Gradually prune weights in dense and convolutional layers from 10% to 50% over the course of training, starting from epoch 0.\n",
    "\n",
    "\n",
    "rates = [0.4, 0.5, 0.6]\n",
    "idx = 20\n",
    "for r in rates:\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=0.01, final_pruning_factor=r, start_epoch=5, end_epoch=20, frequency=1)\n",
    "    model.fit(X_train_normalized_new, y_train, epochs=num_epochs, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import tensorflow as tf\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    from tensorflow.keras.models import Model\n",
    "    layers = model.layers\n",
    "    second_last_layer_output = layers[-4].output\n",
    "    feature_extractor_model = Model(inputs=model.input, outputs=second_last_layer_output)\n",
    "    train_features = feature_extractor_model.predict(X_train_normalized_new)\n",
    "    test_features = feature_extractor_model.predict(X_test_normalized_new)\n",
    "    \n",
    "    p_train = people_nums[train_idx,:]\n",
    "    p_val = people_nums[val_idx,:]\n",
    "    p_test = people_nums[test_idx,:]\n",
    "    ## Calculate the distance between test person and training person\n",
    "    def euclidean_distance(a, b):\n",
    "        return np.sqrt(np.sum((a - b) ** 2, axis=1))\n",
    "    \n",
    "    distance_dict = {}\n",
    "    for ii in range(len(person_nums)):\n",
    "        if person_nums[ii] == test_person_id[0]:\n",
    "            continue\n",
    "        else:\n",
    "            \n",
    "            ind = np.where(p_train ==person_nums[ii])[0]\n",
    "            tmp_train_features = train_features[ind, :]\n",
    "            distances = np.array([euclidean_distance(train_sample, test_features) for train_sample in tmp_train_features])\n",
    "            print(distances.shape)\n",
    "            average_distances = np.mean(distances, axis=1)\n",
    "    \n",
    "            # Step 4: Find the overall average distance\n",
    "            overall_average_distance = np.mean(average_distances)\n",
    "            distance_dict[person_nums[ii]] = overall_average_distance\n",
    "    \n",
    "    \n",
    "    def normalize_to_weights(distance_dict):\n",
    "        distances = np.array(list(distance_dict.values()))\n",
    "        # Handle the case where a distance is zero to avoid division by zero\n",
    "        distances = np.clip(distances, a_min=1e-10, a_max=None)\n",
    "        weights = 1 / distances\n",
    "        normalized_weights = weights\n",
    "        # normalized_weights = weights / sum(weights)\n",
    "        # print(sum(weights))\n",
    "        # print(sum(normalized_weights))\n",
    "        # Assign the normalized weights back to the dictionary\n",
    "        normalized_weight_dict = dict(zip(distance_dict.keys(), normalized_weights))\n",
    "        return normalized_weight_dict\n",
    "    def scale_dict_values(my_dict):\n",
    "        scaled_dict = my_dict.copy()\n",
    "        min_val = min(scaled_dict.values())\n",
    "        max_val = max(scaled_dict.values())\n",
    "        \n",
    "        for key in scaled_dict:\n",
    "            scaled_dict[key] = 1 + 9 * (scaled_dict[key] - min_val) / (max_val - min_val)\n",
    "        \n",
    "        return scaled_dict\n",
    "    weights_dict = normalize_to_weights(distance_dict)\n",
    "    weights_dict = scale_dict_values(weights_dict)\n",
    "    print(weights_dict)\n",
    "    \n",
    "    w_train = np.zeros_like(p_train)\n",
    "    for i in range(len(w_train)):\n",
    "        w_train[i] = weights_dict[int(p_train[i])]\n",
    "    \n",
    "    w_train = np.squeeze(w_train)\n",
    "    \n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,sample_weight= w_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])\n",
    "    import sys\n",
    "    sys.path.append('..')\n",
    "    # for layer in model.layers[-4:]:\n",
    "    #     layer.trainable = False\n",
    "    # model = individual_model.individual_model(X_train_normalized)\n",
    "    model_name = './checkpoints/unknown_person_baseline_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    model = tf.keras.models.load_model(model_name)\n",
    "    model_name = './checkpoints/unknown_person_baseline_2_p'+str(test_person_id[0])+'_'+str(idx)+'.h5'\n",
    "    idx += 1\n",
    "    checkpoint = ModelCheckpoint(model_name, monitor='val_loss', mode='min', save_best_only=True, verbose=1)\n",
    "    pruning_callback = PruningCallback(initial_pruning_factor=r, final_pruning_factor=r, start_epoch=1, end_epoch=20, frequency=1)\n",
    "    \n",
    "    history = model.fit(x=X_train_normalized_new, y=y_train,epochs=20, batch_size=1024, validation_data=(X_val_normalized_new, y_val), callbacks=[pruning_callback, checkpoint])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f71b59e4-6581-4ea8-994d-a02a6654ab6b",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
